functions.opengraph.php

```diff
@@ -94,6 +94,24 @@ function jetpack_og_tags() {
 			$tags['profile:first_name'] = get_the_author_meta( 'first_name', $author->ID );
 			$tags['profile:last_name']  = get_the_author_meta( 'last_name', $author->ID );
 		}
+	} elseif ( is_archive() ) {
+		$tags['og:type'] 		= 'website';
+		$tags['og:title']		= wp_get_document_title();
+		$tags['og:site_name']	= get_bloginfo( 'name' );
+
+		$archive = get_queried_object();
+		if ( ! empty( $archive ) ) {
+			if ( is_category() || is_tag() || is_tax() ) {
+				$tags['og:url'] 		= get_term_link( $archive->term_id, $archive->$taxonomy );
+				$tags['og:description']	= $archive->description;
+			} elseif ( is_post_type_archive() ) {
+				$tags['og:url'] 		= get_post_type_archive_link( $archive->name );
+				$tags['og:description']	= $archive->description;
+			}
+		} else { // for date archives and future-proofing
+			$tags['og:url'] 		= home_url( '/' );
```

Comment:
Do we want the date archives to return the home page link? Is that better than not displaying any OG data?

Reply:
Hm, that's a tricky one. OG data will still appear in the head without an og:url but it's certainly far less useful. Reconstructing the correct date URL to include in the data appears to be a bigger fix. So it seems the options are either not outputting OG data for date archives or outputting it with the home page URL, which would be less than accurate. Perhaps it's better to err on the side of accuracy and omit it altogether.